proxy/http/HttpSM.cc

```diff
@@ -7727,6 +7727,7 @@ HttpSM::set_next_state()
     if (t_state.api_next_action == HttpTransact::SM_ACTION_API_SEND_RESPONSE_HDR) {
       handle_api_return();
     } else {
+      do_drain_request_body(t_state.hdr_info.client_response);
```

Comment:
Is there a reason you are doing this within the conditional and not always draining?

Reply:
When I reproduced it in the same way as  [https://github.com/apache/trafficserver/issues/8951](https://github.com/apache/trafficserver/issues/8951) , I did a drain here and the problem was solved.
I don't have a reason not always drain.